Have the tutor card appear directly above the supplemental card on mobile

tjiang11 opened the pull request.

tjiang11 committed:
Have the tutor card appear directly above the supplemental card on mobile.

kevinlee12 commented:
Ran a UI test on this, the tutor card does appear on top of the supplemental card and in left-right orientation on desktop. However, it appears that the supplemental card is covering the directions (which is shown in desktop) for the tutor card, see: 
![2017-10-04 22-59-29 16ecaa02](https://user-images.githubusercontent.com/5758256/31212857-e3315600-a957-11e7-855a-574641c2fa32.png)


Tested this with my iPhone (portrait and landscape).

kevinlee12 commented:
Just to clarify, it is specifically the "Click on image" that is being covered, this is normally shown on Desktop and the older version of mobile view.

seanlip commented:
Thanks, @kevinlee12!

Btw, in your screenshot, is the bottom right of the image clickable (or is it possible to scroll up further)? Was wondering whether it gets overlapped by the attribution guide, since that might be an issue. If there's a problem here, I think the correct fix is probably to allow scrolling up further than what's shown.

seanlip commented on core/templates/dev/head/pages/exploration_player/conversation_skin_directive.html:
```diff
@@ -383,7 +375,7 @@
 
     .conversation-skin-tutor-card-container {
-      position: absolute;
+      box-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 2px 4px rgba(0, 0, 0, 0.23);
```
Just wondering -- where did this come from? Was it an omission before?

seanlip replied:
Oh wait, ignore that. I think I get it.

tjiang11 replied:
Ah, without it, there's a noticeable disparity in the shadows between the top tutor card and the supplemental card. There should probably be a more cohesive way to address this though since I had to copy this over from the tutor card.

seanlip commented on core/templates/dev/head/pages/exploration_player/tutor_card_directive.html:
```diff
@@ -103,11 +102,11 @@
         <i class="material-icons md-18" style="position: relative; top: 3px;">&#xE5C8;</i>
       </div>
-      <div ng-if="isViewportNarrow()">
+<!--       <div ng-if="isViewportNarrow()">
```
?

Please either remove fully or uncomment. I suspect this is related to @kevinlee12's UI comment.

tjiang11 replied:
done

seanlip commented on extensions/interactions/CodeRepl/CodeRepl.html:
```diff
@@ -41,12 +41,6 @@
     width: 100%;
   }
-
-  @media screen and (max-width: 959px) {
```
Hm, not objecting to these changes (this one and the one in GraphInput.html), but could you explain why they were made?

tjiang11 replied:
With these still in place, the graph or code interaction will appear below the tutor card but much fatter and it just looks awkward because there's empty space around the tutor card

kevinlee12 commented:
The attribution guide does overlap the right corner (so you can click on the icon instead of the image), it is possible to scroll down by another couple of pixels (but doesn't change much).

tjiang11 commented:
Thanks for the review @kevinlee12 and @seanlip ! I'll try to see if I can figure it out but I'm somewhat busy now until the end of the week, so I'm not positive I can make it in time for the release cut.

tjiang11 committed:
Add non-clickable instructions to mobile.

tjiang11 commented:
Still need to fix the attribution guide overlapping. Another issue I noticed is that every so often the response on the supplemental card appears too far downwards to the point you can't really see it--tested on the graph interaction.

seanlip commented:
Hm, that seems broken. I wonder if we can add a div at the bottom that shows up only on mobile and whose only purpose is to just take up some vertical space (maybe 1/3-1/2 of a screen's worth).

By the "response on the supplemental card", do you mean the little card that pops in at the bottom right? Maybe we need to style that so that, on mobile, it appears at some minimum distance above the bottom edge of the screen.

tjiang11 committed:
Prevent overlapping with attribute guide on mobile; Set fixed supplemental response card on mobile.

tjiang11 commented:
Went with your suggestion and added an empty div on mobile.

Regarding the supplemental card response I added a condition so that the position is set to fixed on mobile, though I'm not sure what the use of the fixed position was in the first place.

seanlip commented:
Re second question, I did a git blame: looks like it might have been introduced in #2693. Might be worth double-checking that but if everything works well, then whatever you do should be fine.

Also, @kevinlee12 -- mind taking a look at this from a UI perspective, please? The code LGTM.

kevinlee12 commented:
The "Click on Image" button being covered issue is fixed on the Image Interaction and "Create a graph" button Graph interaction. Also looks good for the other single card interactions. So from the UI, it's good @seanlip @tjiang11!

seanlip commented:
Interesting, the e2e embedding tests are failing. I just restarted them, let's see what happens.

seanlip commented:
Hm, nope, still failing. @tjiang11 -- any idea why? Possibly related to this change?

tjiang11 commented:
Sorry, been poking around but haven't managed to figure much out yet...

seanlip commented:
Hi @tjiang11, I took a look and I think I have some clues!

Try starting up a local server and following the instructions to embed an exploration using an iframe -- load any exploration and click the embed button in the bottom-right corner, then create an HTML file with that code snippet. You'll see that the learner view isn't loading at all in the iframe.

As it turns out, ConversationSkinDirective.js is also used in the embed case, but [with a different HTML file](https://github.com/tjiang11/oppia/blob/1a69a7bf51d66f31e5a2d6551162a45662fdabfb/core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js#L239). That HTML file probably also shares some of the functionality you removed from the JS file. I think that's likely to be the root cause of the problem, and you might either need to edit the conversation_skin_embed_directive HTML file (probably preferable) or reinstate the removed functions in the JS file.

Does that help? If not I can try looking at it more.

tjiang11 committed:
Modify embedded conversation directive to match changes.

tjiang11 commented:
Ah yes that was it, thanks @seanlip !

tjiang11 committed:
Use CSS to prevent overlapping with attribution guide.

seanlip reviewed (approved):
Hooray, thanks @tjiang11! LGTM.
